components/feature/session/src/main/java/mozilla/components/feature/session/SessionUseCases.kt

```diff
@@ -28,7 +28,14 @@ class SessionUseCases(
      * Contract for use cases that load a provided URL.
      */
     interface LoadUrlUseCase {
-        fun invoke(url: String, flags: LoadUrlFlags = LoadUrlFlags.none())
+        /**
+         * Loads the provided URL using the currently selected session.
+         */
+        fun invoke(
+            url: String,
+            flags: LoadUrlFlags = LoadUrlFlags.none(),
+            additionalHeaders: Map<String, String>? = null
```

Comment:
FYI: This may need a patch in Fenix too since I think we sometimes implement this interface directly in Fenix.